packages/react/dialog/src/Dialog.tsx

```diff
@@ -127,13 +128,20 @@ DialogTrigger.displayName = TRIGGER_NAME;
 
 const PORTAL_NAME = 'DialogPortal';
 
+const [DialogPortalProvider, useDialogPortalContext] = createDialogContext(PORTAL_NAME, {
+  isInsidePortal: false,
+});
+
 type PortalProps = React.ComponentPropsWithoutRef<typeof UnstablePortal>;
 interface DialogPortalProps extends PortalProps {}
 
 const DialogPortal: React.FC<DialogPortalProps> = (props: ScopedProps<DialogPortalProps>) => {
-  const { __scopeDialog, ...portalProps } = props;
-  const context = useDialogContext(PORTAL_NAME, __scopeDialog);
-  return context.open ? <UnstablePortal {...portalProps} /> : null;
+  const { __scopeDialog, children } = props;
+  return (
+    <DialogPortalProvider scope={__scopeDialog} isInsidePortal>
+      {children}
+    </DialogPortalProvider>
+  );
 };
```

Comment:
This has actually got me wondering if we need this part now?

It seems we could just have a `portalled` prop on the `Dialog.Root` part that we pass through context instead because the portal doesn't render any DOM anymore.

Reply:
But then how would we allow custom containers?

Reply:
Oh yeah, I've broken that part here 😅 forgot about that. I've just updated this to make sure props get forwarded.